src/main/scala/chiseltest/formal/FirrtlUtils.scala

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: Apache-2.0
+package chiseltest.formal
+
+import chisel3.util.log2Ceil
+import firrtl._
+
+/** firrtl utility functions used in our passes.
+  * At some point, these could be moved into the firrtl repo.
+  */
+private object FirrtlUtils {
+  def plusOne(e: ir.Expression): ir.Expression = {
+    val width = e.tpe.asInstanceOf[ir.UIntType].width.asInstanceOf[ir.IntWidth].width
+    val addTpe = ir.UIntType(ir.IntWidth(width + 1))
+    val add = ir.DoPrim(PrimOps.Add, List(e, ir.UIntLiteral(1, ir.IntWidth(width))), List(), addTpe)
+    ir.DoPrim(PrimOps.Bits, List(add), List(width - 1, 0), e.tpe)
+  }
+
+  def findClockAndReset(m: ir.Module): (ir.Reference, ir.Reference) = {
+    val clock = m.ports
+      .find(_.name == "clock")
+      .getOrElse(
+        throw new RuntimeException(s"[${m.name}] Expected module to have a port named clock!")
+      )
+    val reset = m.ports
+      .find(_.name == "reset")
+      .getOrElse(
+        throw new RuntimeException(s"[${m.name}] Expected module to have a port named reset!")
+      )
+    (ir.Reference(clock).copy(flow = SourceFlow), ir.Reference(reset).copy(flow = SourceFlow))
```

Comment:
I don't love this, I'd rather use analysis to find the clock and reset based on what registers use but I'll accept it for this initial version.

Reply:
I was thinking about that, but since most of `chiseltest` does not really support multi-clock or multi-reset circuits, including our formal verification infrastructure, I don't think it makes sense to solve this problem in this PR.
Once we move `past` to Chisel, we can think a bit more about how to support multi-clock scenarios. However, I do think we need a comprehensive clock domain analysis pass before we can make this a reality, since we would want to be able to catch errors where the `past` delay registers are on a different clock domain then the assertion. That would require a global analysis since two clock inputs could be connected to the same or different clock domain outside of the current module. I do have a [PR for the prototype of a global clock and reset analysis](https://github.com/chipsalliance/firrtl/pull/2252). But to solve the problem, I think we would first have to tighten up our type system a bit and maybe restrict the an analysis to clock signals and _async_ resets.